src/libcore/iter.rs

```diff
@@ -4093,12 +4178,15 @@ pub struct Fuse<I> {
     done: bool
 }
 
+#[unstable(feature = "fused", reason = "recently added", issue = "0")]
+impl<I> FusedIterator for Fuse<I> where I: Iterator {}
+
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<I> Iterator for Fuse<I> where I: Iterator {
     type Item = <I as Iterator>::Item;
 
     #[inline]
-    fn next(&mut self) -> Option<<I as Iterator>::Item> {
+    default fn next(&mut self) -> Option<<I as Iterator>::Item> {
```

Comment:
Unless I'm mistaken, that doesn't apply here (yet) because we don't have the lattice rule. That is, a specialized implementation would have to have the bound `I: FusedIterator + Something` and the implementation on `FusedIterator` isn't marked `default`. However, that's a good point. I'll make a note of this on the specialization issue.